src/util/std_expr.h

```diff
@@ -4654,6 +4657,23 @@ class class_method_descriptor_exprt : public nullary_exprt
   }
 };
 
+inline void validate_expr(const class class_method_descriptor_exprt &value)
```

Comment:
I’m not sure what the `class` keyword is doing here. Seems redundant?

Reply:
It isn't redundant. It is an inline forward declaration of the `class_method_descriptor_exprt` class. The forward declaration of this `validate_expr` function needs to be before the definition of the `class_method_descriptor_exprt` class because it is used from inside the classes constructor. The class also needs to be forward declared in order for the function forward declaration to be valid. So the two forward declarations break a cyclic dependency problem and can't be removed without either re-introducing the problem, or defining the constructor out of line instead of in-line. I could remove the keyword `class` from this line of code and put `class class_method_descriptor_exprt;` on the proceeding line, but this was a little more succinct.

Reply:
FWIW coding guidelines say forward declarations should be put at the top of the file with the includes

Reply:
Ah, I'll try and remember for next time.